common/unsafe/src/main/java/org/apache/spark/sql/catalyst/util/CollationFactory.java

```diff
@@ -1193,6 +1193,10 @@ public static String[] getICULocaleNames() {
     return Collation.CollationSpecICU.ICULocaleNames;
   }
 
+  public static UTF8String applyTrimmingPolicy(UTF8String input, int collationId) {
+    return Collation.CollationSpec.applyTrimmingPolicy(input, collationId);
+  }
```

Comment:
perhaps we should reconsider redirecting a 1-line method call

no hard opinion, but doesn't seem useful at first glance

Reply:
In this case I had to because CollationSpec is not visible (private class) outside of the CollationFactory.